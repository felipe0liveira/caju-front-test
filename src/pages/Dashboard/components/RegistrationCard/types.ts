import { Registration, RegistrationStatus } from "~/types/registration";

export interface RegistrationCardProps {
  data: Registration;
  onDelete: (registration: Registration) => void;
  onUpdate: (registration: Registration, status: RegistrationStatus) => void;
}
