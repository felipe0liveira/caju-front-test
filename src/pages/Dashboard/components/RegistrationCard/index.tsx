import { ButtonSmall } from "~/components/Buttons";
import * as S from "./styles";
import {
  HiOutlineMail,
  HiOutlineUser,
  HiOutlineCalendar,
  HiOutlineTrash,
} from "react-icons/hi";
import { RegistrationStatus } from "~/types/registration";
import { RegistrationCardProps } from "./types";

const RegistrationCard = (props: RegistrationCardProps) => {
  const handleUpdateStatus = (status: RegistrationStatus) => {
    props.onUpdate(props.data, status);
  };

  return (
    <S.Card>
      <S.IconAndText>
        <HiOutlineUser />
        <h3>{props.data.employeeName}</h3>
      </S.IconAndText>
      <S.IconAndText>
        <HiOutlineMail />
        <p>{props.data.email}</p>
      </S.IconAndText>
      <S.IconAndText>
        <HiOutlineCalendar />
        <span>{props.data.admissionDate}</span>
      </S.IconAndText>
      <S.Actions>
        <S.Status>
          {props.data.status === "REVIEW" && (
            <ButtonSmall
              bgcolor="rgb(155, 229, 155)"
              onClick={() => handleUpdateStatus("APPROVED")}
            >
              Aprovar
            </ButtonSmall>
          )}
          {props.data.status === "REVIEW" && (
            <ButtonSmall
              bgcolor="rgb(255, 145, 154)"
              onClick={() => handleUpdateStatus("REPROVED")}
            >
              Reprovar
            </ButtonSmall>
          )}
          {props.data.status !== "REVIEW" && (
            <ButtonSmall
              bgcolor="#ff8858"
              onClick={() => handleUpdateStatus("REVIEW")}
            >
              Revisar novamente
            </ButtonSmall>
          )}
        </S.Status>

        <HiOutlineTrash onClick={() => props.onDelete(props.data)} />
      </S.Actions>
    </S.Card>
  );
};

export default RegistrationCard;
