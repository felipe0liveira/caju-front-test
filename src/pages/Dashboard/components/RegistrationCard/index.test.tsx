import RegistrationCard from ".";
import { render } from "@testing-library/react";
import { describe, test, expect } from "@jest/globals";

describe("<RegistrationCard />", () => {
  test("Should match snapshot", () => {
    const component = render(
      <RegistrationCard
        data={{
          admissionDate: "22/10/2023",
          email: "luiz@caju.com.br",
          employeeName: "Luiz Filho",
          status: "APPROVED",
          cpf: "56642105087",
          id: "3",
        }}
        onDelete={() => {}}
        onUpdate={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
