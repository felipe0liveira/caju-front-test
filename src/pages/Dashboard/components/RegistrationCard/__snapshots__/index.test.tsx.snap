// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegistrationCard /> Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-eqUzNf cOgQpI"
      >
        <div
          class="sc-fqkwJk kWKTdg"
        >
          <svg
            aria-hidden="true"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 7a4 4 0 11-8 0 4 4 0 018 0zM12 14a7 7 0 00-7 7h14a7 7 0 00-7-7z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <h3>
            Luiz Filho
          </h3>
        </div>
        <div
          class="sc-fqkwJk kWKTdg"
        >
          <svg
            aria-hidden="true"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 8l7.89 5.26a2 2 0 002.22 0L21 8M5 19h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v10a2 2 0 002 2z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <p>
            luiz@caju.com.br
          </p>
        </div>
        <div
          class="sc-fqkwJk kWKTdg"
        >
          <svg
            aria-hidden="true"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 7V3m8 4V3m-9 8h10M5 21h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v12a2 2 0 002 2z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <span>
            22/10/2023
          </span>
        </div>
        <div
          class="sc-dcJtft ixtngc"
        >
          <div
            class="sc-iGgVNO iaJeLC"
          >
            <button
              bgcolor="#ff8858"
              class="sc-gEvDqW hglJA-D"
            >
              Revisar novamente
            </button>
          </div>
          <svg
            aria-hidden="true"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-eqUzNf cOgQpI"
    >
      <div
        class="sc-fqkwJk kWKTdg"
      >
        <svg
          aria-hidden="true"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 7a4 4 0 11-8 0 4 4 0 018 0zM12 14a7 7 0 00-7 7h14a7 7 0 00-7-7z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <h3>
          Luiz Filho
        </h3>
      </div>
      <div
        class="sc-fqkwJk kWKTdg"
      >
        <svg
          aria-hidden="true"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 8l7.89 5.26a2 2 0 002.22 0L21 8M5 19h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v10a2 2 0 002 2z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <p>
          luiz@caju.com.br
        </p>
      </div>
      <div
        class="sc-fqkwJk kWKTdg"
      >
        <svg
          aria-hidden="true"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 7V3m8 4V3m-9 8h10M5 21h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v12a2 2 0 002 2z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <span>
          22/10/2023
        </span>
      </div>
      <div
        class="sc-dcJtft ixtngc"
      >
        <div
          class="sc-iGgVNO iaJeLC"
        >
          <button
            bgcolor="#ff8858"
            class="sc-gEvDqW hglJA-D"
          >
            Revisar novamente
          </button>
        </div>
        <svg
          aria-hidden="true"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
